### ISSUE TITLE
Error: Missing html/text field on Linter Response

### ISSUE THREAD

**Issue by camara-tech** [fsievers/linter-perl-check-syntax#2]
Stack Trace:

```
Error: Missing html/text field on Linter Response
    at /Users/italy-portugal/.atom/packages/linter/lib/validate.coffee:39:19
    at Array.forEach (native)
    at Object.module.exports.Validate.messages (/Users/italy-portugal/.atom/packages/linter/lib/validate.coffee:26:14)
    at MessageRegistry.set (/Users/italy-portugal/.atom/packages/linter/lib/message-registry.js:36:16)
    at /Users/italy-portugal/.atom/packages/linter/lib/linter.coffee:37:17
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:125:28)
    at /Users/italy-portugal/.atom/packages/linter/lib/linter-registry.js:66:26
```

on MacOSX, Atom 1.8.0, Linter 1.11.14

**Comment by fsievers** [comment]
Can you supply a minimal perl script to reproduce the error? At the moment, I'm not able to reprocude this.

**Comment by DanFreed** [comment]
I'm seeing the same issue, but haven't isolated what is triggering it.  Interesting however that I can take the exact same script, copy and past it to a new atom window save it to a new name and the error will not occur.

Here is my stack dump:

Error: Missing html/text field on Linter Response
    at /Users/dfreed/.atom/packages/linter/lib/validate.coffee:39:19
    at Array.forEach (native)
    at Object.module.exports.Validate.messages (/Users/dfreed/.atom/packages/linter/lib/validate.coffee:26:14)
    at MessageRegistry.set (/Users/dfreed/.atom/packages/linter/lib/message-registry.js:36:16)
    at /Users/dfreed/.atom/packages/linter/lib/linter.coffee:37:17
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom Beta.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom Beta.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:129:28)
    at /Users/dfreed/.atom/packages/linter/lib/linter-registry.js:66:26

**Comment by DanFreed** [comment]
It only seems to occur when warnings are enabled.
I've included a screen shot from the debugger that I hope will help with this.

Also here is the result string that results in an empty message being put on to the messages list:
`Subroutine Epic::OSLib::getcwd redefined at /Users/dfreed/sourcecode/ssystems/development/fixes/445666/EpicLib/bin/Epic/OSLib.pm line 73. Constant subroutine Epic::OSLib::ENOENT redefined at /usr/local/epic/Epic2016R1/lib/perl/Exporter.pm line 66. at /Users/dfreed/sourcecode/ssystems/development/fixes/445666/EpicLib/bin/Epic/OSLib.pm line 80. Reference found where even-sized list expected. "my" variable $wd_status masks earlier declaration in same scope. Variable "$return_all" will not stay shared. False [] range "\s-" in regex; marked by <-- HERE in m/([\w\s- <-- HERE :]+),*\s*([\w\s:]*)/. Useless use of a constant ("An unknown execution error occur"...) in void context. "my" variable $epic_conf masks earlier declaration in same scope. Use of uninitialized value $Epic::L10N::en_us::integchk_common::duplicate in concatenation (.) or string. Use of uninitialized value $path in -f. Use of uninitialized value $path in pattern match (m//). Use of uninitialized value $path in concatenation (.) or string. Useless use of a constant ("STRING") in void context. Useless use of private variable in void context. "my" variable $nofail masks earlier declaration in same scope. "my" variable $wrappertype masks earlier declaration in same scope. Can't load '/usr/local/epic/Epic2016R1/lib/perl/arch/auto/Net/SSLeay/SSLeay.bundle' for module Net::SSLeay: dlopen(/usr/local/epic/Epic2016R1/lib/perl/arch/auto/Net/SSLeay/SSLeay.bundle, 1): Symbol not found: _X509_EXTENSION_free Referenced from: /usr/local/epic/Epic2016R1/lib/perl/arch/auto/Net/SSLeay/SSLeay.bundle Expected in: flat namespace in /usr/local/epic/Epic2016R1/lib/perl/arch/auto/Net/SSLeay/SSLeay.bundle at /usr/local/epic/Epic2016R1/lib/perl/arch/DynaLoader.pm line 197.  at /Users/dfreed/sourcecode/ssystems/development/fixes/445666/EpicLib/bin/Epic/ESMP/Public.pm line 68. Compilation failed in require at /Users/dfreed/sourcecode/ssystems/development/fixes/445666/EpicLib/bin/Epic/ESMP/Public.pm line 68. BEGIN failed--compilation aborted at /Users/dfreed/sourcecode/ssystems/development/fixes/445666/EpicLib/bin/Epic/ESMP/Public.pm line 68. Compilation failed in require at - line 9. BEGIN failed--compilation aborted at - line 9. 
`

Here is what the empty match looks like in the debugger:
`match
:
Array[4]
0
:
" at /Users/dfreed/sourcecode/ssystems/development/fixes/445666/EpicLib/bin/Epic/OSLib.pm line 80"
1
:
""
2
:
"/Users/dfreed/sourcecode/ssystems/development/fixes/445666/EpicLib/bin/Epic/OSLib.pm"
3
:
"80"
`

![screenshot_32](https://cloud.githubusercontent.com/assets/7361751/18998237/713e8b76-86fd-11e6-88aa-1d4b4fb3ee08.png)

**Comment by mgruner** [comment]
I also see this one, when the linter chokes on the following output (because of a missing include path):

```
perl -cw Kernel/TidyAll/Plugin/OTRS/PO/Lint.pm 
Base class package "TidyAll::Plugin::OTRS::Base" is empty.
    (Perhaps you need to 'use' the module which defines that package first,
    or make that module available in @INC (@INC contains: /Library/Perl/5.18/darwin-thread-multi-2level /Library/Perl/5.18 /Network/Library/Perl/5.18/darwin-thread-multi-2level /Network/Library/Perl/5.18 /Library/Perl/Updates/5.18.2/darwin-thread-multi-2level /Library/Perl/Updates/5.18.2 /System/Library/Perl/5.18/darwin-thread-multi-2level /System/Library/Perl/5.18 /System/Library/Perl/Extras/5.18/darwin-thread-multi-2level /System/Library/Perl/Extras/5.18 .).
 at Kernel/TidyAll/Plugin/OTRS/PO/Lint.pm line 18.
BEGIN failed--compilation aborted at Kernel/TidyAll/Plugin/OTRS/PO/Lint.pm line 18.
```

It would be really great if this could be fixed. Source code for this example is at https://github.com/OTRS/otrscodepolicy.

**Comment by mgruner** [comment]
Thanks a lot, works fine!
